src/cli/services/linter/utils/deduplicateResults.ts

```diff
@@ -1,26 +1,28 @@
-import { IRange } from '@stoplight/types/dist';
+import { resultsComparer } from '../../../../formatters/utils/sortResults';
 import { IRuleResult } from '../../../../types';
 
-type Dictionary<T, K extends PropertyKey> = { [key in K]: T };
+export const deduplicateResults = (results: IRuleResult[]): IRuleResult[] => {
+  const filtered: IRuleResult[] = [];
 
-const ARTIFICIAL_ROOT = Symbol('root');
+  const howMany = results.length;
 
-const serializeRange = ({ start, end }: IRange) => `${start.line}:${start.character}:${end.line}:${end.character}`;
-const getIdentifier = (result: IRuleResult) => `${result.path.join('/')}${result.code}${serializeRange(result.range)}`;
+  if (howMany < 2) {
+    return [...results];
+  }
 
-export const deduplicateResults = (results: IRuleResult[]) => {
-  const seen: Dictionary<Dictionary<string, string>, symbol> = {};
+  const sorted = results.sort(resultsComparer);
```

Comment:
I guess you don't want to mutate `results`.
```suggestion
  const sorted = [...results].sort(resultsComparer);
```

Reply:
@P0lip Dammit! Thanks for this.